torch/lib/TensorLib/code_template.py

```diff
@@ -0,0 +1,69 @@
+import re
+
+# match $idenifier or ${identifier} and replace with value in env
+# If this identifier is at the beginning of whitespace on a line
+# and its value is a list then it is treated as
+# block subsitution by indenting to that depth and putting each element
+# of the list on its own line
+# if the identifier is on a line starting with non-whitespace and a list
+# then it is comma separated ${,foo} will insert a comma before the list
+# if this list is not empty and ${foo,} will insert one after.
+
+
+class CodeTemplate(object):
```

Comment:
This template logic is used to take ${Type|Tensor|Storage} --> (e.g.) CPUByteType, GPULongTensor, CPUIntStorage